### PR TITLE
App: Update user profile and admin links

### PR DESCRIPTION
**Pull request by philipp-spiess**
Part of #48402

This PR is a first stab at #48402 and specifically the items outlined [in this doc](https://docs.google.com/document/d/1DH-hlcTMs5GOUXPcqy9nJ9RIgP4gmlrMn6GD5pxq5Bc/edit).

Here's a full recap of the changes starting from the user dropdown menu (Changed pages are marked bold):

- **Settings**
  - **Profile**: _Tab was removed_
  - Settings
    - Account
      - Settings: kept
      - **Profile**: removed
      - Batch Changes: kept
      - Executor secrets: kept
      - **Emails:** removed
      - Access tokens: kept
      - **Account security**: removed
      - **Product research:** updated, see screenshot
      - **Permissions**: removed
      - Event log: kept
    - **Your organizations**: removed
    - Other actions: 
      - API console: kept
      - Site admin: kept
  - Saved searches: kept
  - **Batch Changes**: ~~⚠️ _Tab was removed_ but it's unclear to me if this change makes sense? How are batch changes going to work in App?~~ kept, see comments below
- Saved searches: kept
- Theme: kept
- Site admin
  - Analytics
    - Overview: ⚠️ unclear, see below
    - Search, Code navigation, Users, Insights, Batch changes, Notebooks, Extensions: kept
    - **Feedback survey:** removed
  - Configuration
    - Site configuration: kept 
    - Global settings: kept
  - **Feature flags**: removed
  - License: ⚠️ unclear, see below 
  - Repositories
    - Manage code hosts: kept
    - Repositories: kept
    - Incoming webhooks: kept 
  - Code graph
    - Precise indexes: kept
    - Configuration: kept
    - Inference: kept
  - **Users & auth**: removed
    - **Users**: removed
    - **Organizations**: removed
    - **Access tokens**: removed
  - Maintenance
    - Updates: ⚠️ unclear, see below
    - Documentation: kept
    - Pings: ⚠️ unclear, see below
    - **Report a bug**: link to repo updates
    - Migrations: kept
    - Instrumentation: kept
    - Monitoring: kept    
    - Tracing: kept
    - Outbound requests: kept
    - Slow requests: kept
    - Background jobs: kept
    - Code Insights jobs: kept
  - Batch changes
    - **Settings:** ~~removed ⚠️ it's unclear to me if this change makes sense? How are batch changes going to work in App?~~ kept, see comments below
    - Batch specs: kept
    - **Incoming webhooks**: ~~removed ⚠️ it's unclear to me if this change makes sense? How are batch changes going to work in App?~~ kept, see comments below
    - Outgoing webhooks: kept
  - API console: kept
- **Help**: Changed, now called Documentation
- **Feedback**: Changed, links to Settings → Settings → Product research now
- Keyboard shortcuts: kept
- Browser extension: kept

## Next steps

- Investigate why docs link did not work for @taylorsperry (they work fine on dev)
- Admin > Analytics > Overview: How do we want to change the language
- Admin > License: What do we want to show here?
- Admin > Maintenance > Updates: What do we want to show here?
- Admin > Maintenance > Pings: @marekweb do you plan on upgrading this after the analytics changes are done?

## Test plan

Screenshot 2023-03-07 at 14 39 58
Screenshot 2023-03-07 at 14 00 42
Screenshot 2023-03-07 at 13 57 17

## App preview:

- [Web](https://sg-web-ps-app-menu.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

**Comment by slimsag** [comment]
Thanks for this! I'd highly suggest we lean towards merging any version of this sooner rather than later, and then continue to improve it after (rather than getting it perfect the first go.) PRs like this have a habbit of getting stale quick and difficult to land :)

> Batch Changes: ⚠️ Tab was removed but it's unclear to me if this change makes sense? How are batch changes going to work in App?

@taylorsperry was this your intent? I thought the tab would be kept but maybe I've missed something?

> Admin > Maintenance > Updates: What do we want to show here?

I'd suggest we remove this for now, or rename to "Overview" and remove the "Update" panel on the page so it only shows this portion:

image

Update check functionality will not work in App due to a different versioning approach.

File: client/web/src/user/settings/research/ProductResearch.tsx [hunk 1]
```diff
@@ -33,6 +35,38 @@ export const ProductResearchPage: React.FunctionComponent<React.PropsWithChildre
         <>
             <PageTitle title="Product research" />
             <PageHeader headingElement="h2" path={[{ text: 'Product research and feedback' }]} className="mb-3" />
+            {isSourcegraphApp && (
+                <Container className="mb-2">
+                    <Text>Do you have feedback or need help with Sourcegraph App?</Text>
+                    {[
+                        {
+                            content: 'File an issue',
+                            path: 'https://github.com/sourcegraph/app',
```
Comment (slimsag) [inline client/web/src/user/settings/research/ProductResearch.tsx:44]:
```suggestion
                            path: 'https://github.com/sourcegraph/app/issues/new?assignees=&labels=&template=bug_report.md&title='',
```

File: client/web/src/user/settings/research/ProductResearch.tsx [hunk 2]
```diff
@@ -33,6 +35,38 @@ export const ProductResearchPage: React.FunctionComponent<React.PropsWithChildre
...
+                        },
+                        {
+                            content: 'Join our Discord',
+                            path: 'https://about.sourcegraph.com/community',
+                        },
+                        {
+                            content: 'Follow us on Twitter',
+                            path: 'https://twitter.com/sourcegraph',
+                        },
+                        { content: 'Email us', path: 'mailto:taylor.sperry@sourcegraph.com' },
```
Comment (slimsag) [inline client/web/src/user/settings/research/ProductResearch.tsx:55]:
```suggestion
                        { content: 'Email us', path: 'mailto:support@sourcegraph.com' },
```

lol probably best this goes to a catch-all; either that or we can set up a dedicated email support group for it.

**Review by slimsag — APPROVED** [review]

File: client/web/src/user/settings/research/ProductResearch.tsx (hunk 1 above)
Comment (philipp-spiess) [reply client/web/src/user/settings/research/ProductResearch.tsx:44]: Oh I forgot the `/issues/new` part. 😐 

We don't have a bug_report template there yet but I can include it in the URL so we can add one later 👍

**Comment by philipp-spiess** [comment]
@slimsag Yeah I agree with moving fast here. Just let me know if any of this causes troubles, I’m more than happy to fix :)

**Comment by philipp-spiess** [comment]
> > Batch Changes: ⚠️ Tab was removed but it's unclear to me if this change makes sense? How are batch changes going to work in App?
>
> @taylorsperry was this your intent? I thought the tab would be kept but maybe I've missed something?

Yeah I found this confusing as well, I'll revert for now while we figure out how we want this to be ideally.

File: client/web/src/user/settings/research/ProductResearch.tsx (hunk 2 above)
Comment (philipp-spiess) [reply client/web/src/user/settings/research/ProductResearch.tsx:55]: Haha yeah this was part of this issue: https://github.com/sourcegraph/sourcegraph/issues/48397

Although I’m actually removing it now. There's some discussion in Figma that too many options are confusing and I'd rather have this in sync with the header: https://github.com/sourcegraph/sourcegraph/pull/48838

**Comment by marekweb** [comment]
> Investigate why docs link did not work for @taylorsperry (they work fine on dev)

That might be an outdated report, we merged a fix for the docs links in #48374

**Comment by marekweb** [comment]
> Admin > Maintenance > Pings: @marekweb do you plan on upgrading this after the analytics changes are done?

I think we can just remove it for now

File: client/web/src/user/settings/research/ProductResearch.tsx (hunk 1 above)
Comment (slimsag) [reply client/web/src/user/settings/research/ProductResearch.tsx]: Sounds good, yep, I'd say leave it in the URL so we can add one later if it makes sense.

**Pull request merged**